wd-java/src/test/java/com/example/TestUtils.java

```diff
@@ -9,17 +12,50 @@
 public class TestUtils {
     
     static RemoteWebDriver getWebDriver(String username, String accessKey) throws MalformedURLException {
+        // Set capabilities for WebDriver
+        DesiredCapabilities caps = new DesiredCapabilities();
+        caps.setBrowserName("chrome");
+        return new RemoteWebDriver(getDriverUrl(username,accessKey), caps);
+    }
+
+    static AndroidDriver getAndroidEmulatorDriver(String username, String accessKey) throws MalformedURLException {
+        MutableCapabilities caps = getAndroidEmulatorCapabilities();
+        return new AndroidDriver(getDriverUrl(username, accessKey), caps);
+    }
+
+    static AndroidDriver getAndroidDriver(String username, String accessKey) throws MalformedURLException {
+        MutableCapabilities caps = getAndroidCapabilities();
+        return new AndroidDriver(getDriverUrl(username, accessKey), caps);
+    }
+
+    private static URL getDriverUrl(String username, String accessKey) throws MalformedURLException {
         if (username == null || accessKey == null || username.trim().isEmpty() || accessKey.trim().isEmpty()) {
             String err = "Sauce Labs credentials not found. Please set SAUCE_USERNAME and SAUCE_ACCESS_KEY in your environment";
             throw new RuntimeException(err);
         }
         // Can be found at "Driver creation" on https://app.saucelabs.com/user-settings
-        String webDriverUrl = "https://" + username + ":" + accessKey + "@ondemand.saucelabs.com:443/wd/hub";
+        return new URL("https://" + username + ":" + accessKey + "@ondemand.saucelabs.com:443/wd/hub");
+    }
 
-        // Set capabilities for WebDriver
-        DesiredCapabilities caps = new DesiredCapabilities();
-        caps.setBrowserName("chrome");
+    private static MutableCapabilities getAndroidEmulatorCapabilities() {
+        MutableCapabilities caps = new MutableCapabilities();
+        caps.setCapability("platformName", "Android");
+        caps.setCapability("browserName", "Chrome");
+        caps.setCapability("appium:deviceName", "Android GoogleAPI Emulator");
+        caps.setCapability("appium:platformVersion", "14.0");
+        caps.setCapability("appium:automationName", "UiAutomator2");
+        return caps;
+    }
 
-        return new RemoteWebDriver(new URL(webDriverUrl), caps);
+    private static MutableCapabilities getAndroidCapabilities() {
+        MutableCapabilities caps = new MutableCapabilities();
+        caps.setCapability("platformName", "Android");
+        caps.setCapability("browserName", "Chrome");
+        caps.setCapability("appium:deviceName", "Google.*");
```

Comment:
Please select a "specific" device, `Google.*` will return an available device that matches the regex, meaning a pixel 1-8

Reply:
Done